sample_consensus/include/pcl/sample_consensus/impl/lmeds.hpp

```diff
@@ -96,22 +97,36 @@ pcl::LeastMedianSquares<PointT>::computeModel (int debug_verbosity_level)
       ++skipped_count;
       continue;
     }
+    // Move all NaNs in distances to the end
+    const auto new_end = std::partition (distances.begin(), distances.end(), [](double d){return !std::isnan (d);});
+    const auto nr_valid_dists = std::distance (distances.begin (), new_end);
 
-    std::sort (distances.begin (), distances.end ());
     // d_cur_penalty = median (distances)
-    std::size_t mid = sac_model_->getIndices ()->size () / 2;
-    if (mid >= distances.size ())
+    const std::size_t mid = nr_valid_dists / 2;
```

Comment:
I would say it makes more sense to take the middle of only the valid distances, omitting the NaNs. But I guess `mid >= distances.size ()` is unnecessary, since `nr_valid_dists` should always be smaller or equal to `distances.size()`, so testing `nr_valid_dists == 0` alone should be enough, right?

Reply:
The issue was the old impl switching from index to pointcloud as if they are one and the same. See old 111, 112. But you are correct. Testing `nr_valid_dists` seems sufficient